CRM/Campaign/BAO/Campaign.php

```diff
@@ -270,12 +270,12 @@ public static function getPermissionedCampaigns(
 
   /**
    * Is CiviCampaign enabled.
+   * Only used from the Smarty Template.
    * @deprecated
    * @return bool
    */
-  public static function isCampaignEnable(): bool {
-    CRM_Core_Error::deprecatedFunctionWarning('isComponentEnabled');
-    return self::isComponentEnabled();
+  public static function isComponentEnabled(): bool {
+    return CRM_Core_Component::isEnabled('CiviCampaign');
   }
```

Comment:
Everything else in this PR makes sense and looks like it's as-described 👍 . (In theory, sometimes swapping equivalent things might be a problem if it's pre-boot/mid-boot -- but everything seems to be regular/post-boot/business-y code. So 🎉 )

But this part is confusing:

* The PR doesn't say anything about *dropping* the deprecated function. 
* The new method removes the `deprecatedFunctionWarning()` but keeps `@deprecated`
* Grep still finds some things in universe using the old name. (`bugp`, `scheduler`, `fastactivity`, Systopia `campaign`)
* There are some recent/pre-existing references to `isComponentEnabled()` in Smarty templaets -- OK. But that appears to be inherited. So if it's already there, why does it need an override?

Reply:
(tldr: it would be MOP for the bulk of the PR, except this parts seems out-of-character/confusing.)

Reply:
@totten I thought 2 years of noisy deprecation would be enough, but I guess not. Will restore it.

I was overriding it out of an abundance of caution because the fixed version of the CRM_Core_DAO function from https://github.com/civicrm/civicrm-core/pull/30636 is more metadata-driven so not sure how it does in early-bootstrap situations... but it's probably fine.

Reply:
> I was overriding it out of an abundance of caution because the fixed version of the CRM_Core_DAO function from https://github.com/civicrm/civicrm-core/pull/30636 is more metadata-driven so not sure how it does in early-bootstrap situations... but it's probably fine.

Ah, OK.

> I thought 2 years of noisy deprecation would be enough, but I guess not. Will restore it.

Yeah, that is kind of weird/frustrating. I see the warning was released in 5.55 (~21 months ago). Three of those four extensions have had various commits+updates within 2024, so they're not un-maintained. Thinking out loud...

* Maybe the authors aren't recognizing the warnings? 
* Maybe their testing is focused on older versions?
    * (*Each ext has a different `<compatibility>` target, but all of them have targets <5.55. I doubt that many people do full-range testing -- it's easier to test one version and cross-fingers. And that creates a bias in the QA. For some people, that becomes a old-version-bias; for others, it becomes a new-version-bias. My guess: these ext's are maintained with an old-version-bias.*)

Anyway, if you wanted to push through on dropping, you probably could. But IMHO the function is too small to worry about either way. It's maybe useful to keep in mind as an exemplar when discussing maintability.